src/lib/drivers/device/Device.hpp

```diff
@@ -114,6 +114,22 @@ class __EXPORT Device
 	 */
 	virtual int	write(unsigned address, void *data, unsigned count) { return -ENODEV; }
 
+	/**
+	 * Read a register from the device.
+	 *
+	 * @param		The register to read.
+	 * @return		The value that was read.
+	 */
+	virtual uint8_t read_reg(unsigned reg) { return 0; }
```

Comment:
Implemented for POSIX SPI/I2C only because I did not work with other platforms before and don't have corresponding board to verify correctness of code change on other platform. In this round of change, found that read_reg/write_reg were duplicated in quite some drivers. Because I have no way to verify those drivers, so the duplicated code were not removed, instead, marked the API as override. Things would be easier if these 2 APIs were provided at higher level initially. Also as these 2 APIs were already implemented in many drivers, names of the API were not changed.  Another person who has more experience with these driver code could make more systematic changes later.